ts/features/payments/checkout/screens/WalletPaymentPickMethodScreen.tsx

```diff
@@ -197,7 +254,8 @@ const WalletPaymentPickMethodScreen = () => {
             paymentsCreateTransactionAction.request({
               paymentNotices: [
                 { rptId: paymentDetails.rptId, amount: paymentDetails.amount }
-              ]
+              ],
+              onError: handleOnTransactionCreationError
```

Comment:
The content of `handleOnTransactionCreationError` could be easily handled inside the saga. Why did you choose to use a callback?

Reply:
The callback is necessary to display the toast when the error occurs and keep track of the event. If a POST transaction goes in error, it displays an error screen interrupting the payment flow, but as shown in the following figure, only a toast message must be shown.


https://www.figma.com/design/OwkkVcUF05ZAfNkOp51Wwo/Pagamento---Tracking-Strategy?node-id=61-11205&t=listM3JTa8UHAsJB-4

Reply:
Did you know the we can use [IOToast](https://github.com/pagopa/io-app-design-system/blob/522fc789eb330d64baef3b59e36305b2936bb9cd/src/components/toast/useIOToast.ts#L79) to show toast inside sagas and class components?

Reply:
To be honest, I'm not a big fan of adding "display" code within sagas, as I always prefer to isolate business logic from UI logic.

However, if you believe it is more appropriate to add it to the saga and considering the challenging deadline, I have implemented your suggestion in this commit: 872a4a6bb0cb91f66d60e4d8eba0b81e6ee76f22